airflow/models.py

```diff
@@ -375,7 +392,7 @@ def kill_zombies(self, session=None):
                         except subprocess.CalledProcessError:
                             logging.error("Failed to delete pod - it may have already been destroyed.")
 
-                    ti.handle_failure("{} killed as zombie".format(str(ti)))
+                    ti.handle_zombie("{} killed as zombie".format(str(ti)))
```

Comment:
handle_zombie does almost the same thing as handle_failure, but with state ZOMBIED instead of UP_FOR_RETRY